src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/BackgroundDocumentGenerator.cs

```diff
@@ -203,6 +215,11 @@ private async Task TimerTick()
                     {
                         await ProcessDocument(project, document).ConfigureAwait(false);
                     }
+                    catch (FileNotFoundException)
+                    {
+                        // Ignore FileNotFoundExceptions. These can occur when a file was in the middle of being renamed and it dissapears as we're processing it.
```

Comment:
Should we catch or handle any other kinds of exceptions here? Maybe `IOException`?

Reply:
I don't see the harm, sure why not 😄. 

I'm tempted to also swallow `UnauthorizedAccessException` but I'm uncertain if VS threading (or some other VS piece in this call path) utilizes that for other mechanisms other than file access security violations. Do you happen to have any context there?

Reply:
I can't think of a reason why VS threading would do file I/O